flipperdevices/flipperzero-firmware
# [FL-1181] Archive app

# What's new

### Stage 1 Archive app + various

- Archive: sd card file browsing
- Archive: delete, rename and add to favorites
- Dolphin: added "unlocked" message on unlocking
- Core: added furi/common_defines.h 
- minor macros cleanup
- Gui: elements_scrollbar_pos() for fine scrollbar positioning
- Gui: fix scrollbar min height


# Verification 

- Press Down from idle screen to enter archive 
- No extensions provided by apps for now, so no saved keys will be displayed in corresponding tabs
- Switch to rightmost tab "Browse" to browse all files on sd card
- If you feeling adventurous, prepare some files on PC with any of following extentisions - `.ibtn .irda .sub1 .rfid .nfc` and check it out.
- Keys should be placed into corresponding folders - eg: `favorites/`, `ibutton/`, `subone/`, etc. Its possible to access key from any folder on sd card from "Browse" tab.

# Checklist (do not modify)
